extensions/keycloak-admin-client-reactive/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/runtime/ResteasyReactiveClientProvider.java

```diff
@@ -31,30 +34,54 @@ private ClientBuilderImpl registerJacksonProviders(ClientBuilderImpl clientBuild
-                clientBuilder = clientBuilder.register(new JacksonBasicMessageBodyReader(objectMapper));
-            }
+                InstanceHandle<JacksonBasicMessageBodyReader> readerInstance = arcContainer
+                        .instance(JacksonBasicMessageBodyReader.class);
+                if (readerInstance.isAvailable()) {
+                    clientBuilder = clientBuilder.register(readerInstance.get());
+                } else {
+                    objectMapper = getObjectMapper(objectMapper, objectMapperInstance);
+                    clientBuilder = clientBuilder.register(new JacksonBasicMessageBodyReader(objectMapper));
+                }
 
-            InstanceHandle<ClientJacksonMessageBodyWriter> writerInstance = arcContainer
-                    .instance(ClientJacksonMessageBodyWriter.class);
-            if (writerInstance.isAvailable()) {
-                clientBuilder = clientBuilder.register(writerInstance.get());
+                InstanceHandle<ClientJacksonMessageBodyWriter> writerInstance = arcContainer
+                        .instance(ClientJacksonMessageBodyWriter.class);
+                if (writerInstance.isAvailable()) {
+                    clientBuilder = clientBuilder.register(writerInstance.get());
+                } else {
+                    objectMapper = getObjectMapper(objectMapper, objectMapperInstance);
+                    clientBuilder = clientBuilder.register(new ClientJacksonMessageBodyWriter(objectMapper));
+                }
             } else {
-                objectMapper = getObjectMapper(objectMapper, objectMapperInstance);
-                clientBuilder = clientBuilder.register(new ClientJacksonMessageBodyWriter(objectMapper));
+                ObjectMapper newObjectMapper = new ObjectMapper();
+                clientBuilder = clientBuilder.register(new JacksonBasicMessageBodyReader(newObjectMapper))
+                        .register(new ClientJacksonMessageBodyWriter(newObjectMapper));
             }
+
         }
         return clientBuilder;
     }
 
+    // the idea is to only reuse the ObjectMapper if no known customizations would break Keycloak
+    // TODO: in the future we could also look into checking the ObjectMapper bean itself to see how it has been configured
+    private boolean canReuseObjectMapper(ArcContainer arcContainer) {
+        Instance<ObjectMapperCustomizer> customizers = arcContainer.beanManager().createInstance()
+                .select(ObjectMapperCustomizer.class);
+        if (!customizers.isUnsatisfied()) {
+            // ObjectMapperCustomizer can make arbitrary changes, so in order to be safe we won't allow reuse
+            return false;
+        }
+        InstanceHandle<JacksonBuildTimeConfig> jacksonConfig = arcContainer.instance(JacksonBuildTimeConfig.class);
+        if (jacksonConfig.isAvailable()) {
+            return jacksonConfig.get().propertyNamingStrategy.isEmpty();
+        }
+        return true;
```

Comment:
Don't you think it's going to be brittle? My fear is that we might break it when adding new Jackson customization options.
I'm not sure if it's really worth taking the risk, just to avoid creating a second `ObjectMapper` object.

Reply:
I want to avoid creating a second `ObjectMapper` if at all possible.

The other option here is to look and see if there is any `quarkus.jackson...` configuration option set

Reply:
PR updated to follow the approach mentioned above